src/components/marquee-wizard/marquee-wizard.component.spec.ts

```diff
@@ -273,3 +281,108 @@ describe('Marquee wizard with delayed step creation', () => {
+        >
+            <ux-marquee-wizard-step *ngFor="let step of steps"
+                [header]="step.header" [valid]="step.valid"
+                (visitedChange)="visitedChanged()"
+            >
+                <p>{{ step.content }}</p>
+                <button class="toggle-validity-button" (click)="step.valid = !step.valid">Toggle validity</button>
+            </ux-marquee-wizard-step>
+        </ux-marquee-wizard>
+    `
+})
+class MarqueeWizardVisitedChangeTestComponent {
+    steps: WizardStep[] = [
+        {
+            header: '1. First Step',
+            content: 'Content of step 1.',
+        },
+        {
+            header: '2. Second Step',
+            content: 'Content of step 2.',
+        },
+        {
+            header: '3. Third Step',
+            content: 'Content of step 3.',
+        },
+        {
+            header: '4. Fourth Step',
+            content: 'Content of step 4.',
+        }
+    ];
+    stepChanging(_: StepChangingEvent) {}
+    stepChange(_: number) {}
+    onNext(_: number) {}
+    visitedChanged(_: boolean) { }
+}
+
+describe('Marquee wizard with visitedChange event', () => {
+    let component: MarqueeWizardVisitedChangeTestComponent;
+    let fixture: ComponentFixture<MarqueeWizardVisitedChangeTestComponent>;
+    let nativeElement: HTMLElement;
+    let visitedChanged: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [MarqueeWizardModule, CommonModule],
+            declarations: [MarqueeWizardVisitedChangeTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(async () => {
+        fixture = TestBed.createComponent(MarqueeWizardVisitedChangeTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        visitedChanged = spyOn(component, 'visitedChanged');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+        await fixture.whenStable();
+    });
+
+    it('should trigger a visitedChange event when valid modified on the current step, when other steps ahead are visited', async () => {
+        // set step to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // set step to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // jump back to the first step
+        await clickButton(MarqueeWizardSelectors.FirstStep);
+
+        visitedChanged.calls.reset();
+
+        // valid true and no call of visitedChange
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+
+        // valid now false and should trigger visitedChange
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+
+        expect(visitedChanged).toHaveBeenCalledTimes(1);
```

Comment:
We should also check the value that it was called with, so that it is emitting the correct value. Same for the wizard test